src/PollingBlockTracker.test.ts

```diff
@@ -0,0 +1,3024 @@
+          expect(currentBlockNumber).toStrictEqual('0x0');
+        },
+      );
+    });
+  });
+
+  METHODS_TO_ADD_LISTENER.forEach((methodToAddListener) => {
+    describe(`${methodToAddListener}`, () => {
+      describe('"latest"', () => {
+        it('should start the block tracker', async () => {
+          recordCallsToSetTimeout();
+
+          await withPollingBlockTracker(({ blockTracker }) => {
+            blockTracker[methodToAddListener]('latest', EMPTY_FUNCTION);
+
+            expect(blockTracker.isRunning()).toBe(true);
+          });
+        });
+
+        it('should take a listener that is called soon after being added', async () => {
+          recordCallsToSetTimeout();
+
+          await withPollingBlockTracker(
+            {
+              provider: {
+                stubs: [
+                  {
+                    methodName: 'eth_blockNumber',
+                    response: {
+                      result: '0x0',
+                    },
+                  },
+                ],
+              },
+            },
+            async ({ blockTracker }) => {
+              const latestBlockNumber = await new Promise<string>((resolve) => {
+                blockTracker[methodToAddListener]('latest', resolve);
+              });
+              expect(latestBlockNumber).toStrictEqual('0x0');
+            },
+          );
+        });
+
+        it('should take a listener that is called periodically after being added', async () => {
+          const setTimeoutRecorder = recordCallsToSetTimeout({
+            numAutomaticCalls: 1,
+          });
+
+          await withPollingBlockTracker(
+            {
+              provider: {
+                stubs: [
+                  {
+                    methodName: 'eth_blockNumber',
+                    response: {
+                      result: '0x0',
+                    },
+                  },
+                  {
+                    methodName: 'eth_blockNumber',
+                    response: {
+                      result: '0x1',
+                    },
+                  },
+                ],
+              },
+            },
+            async ({ blockTracker }) => {
+              const receivedBlockNumbers: string[] = [];
+
+              await new Promise<void>((resolve) => {
+                setTimeoutRecorder.onNumAutomaticCallsExhausted(resolve);
+
+                blockTracker[methodToAddListener](
+                  'latest',
+                  (blockNumber: string) => {
+                    receivedBlockNumbers.push(blockNumber);
+                  },
+                );
+              });
+
+              expect(receivedBlockNumbers).toStrictEqual(['0x0', '0x1']);
+            },
+          );
+        });
+
+        it('should change the Timeout object created when the poll loop waits for the next iteration such that Node does not have to wait for it to finish before exiting', async () => {
```

Comment:
I found this description difficult to understand. Maybe we could word it less "imperatively", i.e. describe what we want, not the steps we took to get there.

e.g. something like
```suggestion
        it('should not prevent Node from exiting while the polling is active', async () => {
```
Though I'm not confident that is the best way to word that

Reply:
Another test with the same wording here: https://github.com/MetaMask/eth-block-tracker/pull/103/files#diff-59b2763dcfc8532df91ccd2df5bfd3051d465e6eef759260d2430f4921d28c75R1379

Reply:
Adjusted here: [797fdc3](https://github.com/MetaMask/eth-block-tracker/pull/103/commits/797fdc3401f7f2e60754a48b6cd446c294b86bd1)